src/authhelper.cpp

```diff
@@ -0,0 +1,51 @@
+/*
+  authhelper.cpp
+  This file is part of Hotspot, the Qt GUI for performance analysis.
+  Copyright (C) 2016-2021 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+  Author: Lieven Hey <lieven.hey@kdab.com>
+  Licensees holding valid commercial KDAB Hotspot licenses may use this file in
+  accordance with Hotspot Commercial License Agreement provided with the Software.
+  Contact info@kdab.com if any conditions of this licensing are not clear to you.
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <KAuth>
+#include <QFile>
+#include <QObject>
+#include <QProcess>
+
+class AuthHelper : public QObject
+{
+    Q_OBJECT
+public:
+    AuthHelper() = default;
+
+public slots:
+    KAuth::ActionReply elevate(const QVariantMap& args)
+    {
+        const QString script = args[QStringLiteral("script")].toString();
```

Comment:
please don't pass the script here, instead use `Util::findLibexecBinary` or something similar to that, which works for the kauth helper

otherwise this kauth helper is super dangerous, as it could be used to run arbitrary code masquerading as our elevate script utility - one would just pass an arbitrary script as an argument and it would be executed as root...